### PR TITLE
Give Terraform instances access to two roles

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request:
* Modifies the assume role policy for the organization read-only role so that can be assumed by the Terraformer instances added in cisagov/cool-assessment-terraform#132.
* Updates the policy associated with the COOL AMI KMS key to also allow the Terraformer instances to use it for decrypting COOL AMIs.

## 💭 Motivation and context ##

Any Terraformer instances in assessment accounts require access to the organization read-only role, since they need to be able to pull down things like account IDs.  See cisagov/cool-assessment-terraform#132 for more details.

Any Terraformer instances in assessment accounts require access to the COOL AMI KMS key, since they need to be able to launch EC2 instances based on the COOL AMIs .  See cisagov/cool-assessment-terraform#132 for more details.

## 🧪 Testing ##

I successfully applied these changes, and then I successfully leveraged them in cisagov/cool-assessment-terraform#132.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: images/ami_kms_key.tf [hunk 1]
```diff
@@ -116,11 +116,15 @@ data "aws_iam_policy_document" "ami_kms_doc" {
...
+      values = concat([
         for account in data.aws_organizations_organization.cool.accounts :
         "arn:aws:iam::${account.id}:role/ProvisionAccount"
         if length(regexall("^env[0-9]* \\(${local.this_account_type}\\)$|^Playground Legacy \\(${local.this_account_type}\\)$|^Shared Services \\(${local.this_account_type}\\)$", account.name)) > 0
-      ]
+        ], [
+        for account in data.aws_organizations_organization.cool.accounts :
+        "arn:aws:iam::${account.id}:role/Terraformer"
+        if length(regexall("^env[0-9]* \\(${local.this_account_type}\\)$", account.name)) > 0
+      ])
```
Comment (dav3r) [inline images/ami_kms_key.tf:127]: The comment above this block should be updated to mention the new Terraformer role.

File: master/locals.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# ------------------------------------------------------------------------------
+# Evaluate expressions for use throughout this configuration.
+# ------------------------------------------------------------------------------
+locals {
+  # Look up all assessment account IDs from AWS organizations provider
+  assessment_account_ids = [
+    for account in data.aws_organizations_organization.cool.accounts :
+    account.id
+    if length(regexall("^env\\d+", account.name)) > 0
```
Comment (dav3r) [inline master/locals.tf:16]: Are you planning ahead here for the near future when our assessment env account names change from (for example) `env123 (Staging)` to just `env123`?

**Review by dav3r — APPROVED** [review]
Approved with one small request to update a comment. 🎸

File: images/ami_kms_key.tf (hunk 1 above)
Comment (jsf9k) [reply images/ami_kms_key.tf:129]: See commit fd7cc0f1b792ac7f191da69ad3f92dfc793dbcae.

File: master/locals.tf (hunk 2 above)
Comment (jsf9k) [reply master/locals.tf:22]: I updated the comment to mention this.  See commit e001ac8.

Comment (dav3r) [reply master/locals.tf:22]: Good call. 👍

**Review by mcdonnnj — APPROVED** [review]
Makes sense to me ✔

**Pull request merged**